tests/planner/ompl/test_OMPLSimplifier.cpp

```diff
@@ -0,0 +1,102 @@
+
+  // TODO: unique pointers are lost. Is there a better way to go about this?
+  aikido::distance::DistanceMetricPtr _dmetric = aikido::distance::createDistanceMetric(stateSpace);
+  aikido::constraint::SampleablePtr _sampler = aikido::constraint::createSampleableBounds(stateSpace, make_rng()); 
+  aikido::constraint::ProjectablePtr _boundsProjection = aikido::constraint::createProjectableBounds(stateSpace);
+  aikido::constraint::TestablePtr _boundsConstraint = aikido::constraint::createTestableBounds(stateSpace);
+  aikido::constraint::TestablePtr _collConstraint = std::make_shared<MockTranslationalRobotConstraint>(
+        stateSpace, Eigen::Vector3d(-0.1, -0.1, -0.1),
+        Eigen::Vector3d(0.1, 0.1, 0.1));
+
+  auto simplifiedPair = aikido::planner::ompl::simplifyOMPL(
+      stateSpace, interpolator, std::move(_dmetric), std::move(_sampler), 
+      std::move(_collConstraint), std::move(_boundsConstraint), std::move(_boundsProjection),
+      0.1, 5.0, 10, traj);
+  auto simplifiedTraj = std::move(simplifiedPair.first);
+  bool shorten_success = simplifiedPair.second;
+  
+// Test that the start and goal positions do not change
+//=============================================================================
+  // Check the first waypoint
+  auto s0 = stateSpace->createState();
+  simplifiedTraj->evaluate(0, s0);
+  auto r0 = s0.getSubStateHandle<R3>(0);
+  EXPECT_TRUE(r0.getValue().isApprox(startPose));
+
+  // Check the last waypoint
+  simplifiedTraj->evaluate(simplifiedTraj->getDuration(), s0);
+  r0 = s0.getSubStateHandle<R3>(0);
+  EXPECT_TRUE(r0.getValue().isApprox(goalPose));
+
+// Test that the boolean returned is in agreement with shortening process
+//=============================================================================
+  auto stateCurrent = stateSpace->createState();
+  auto stateNext = stateSpace->createState();
+
+  double trajDistance = 0.0;
+  for(size_t i = 0; i < traj->getDuration(); ++i)
+  {
+    traj->evaluate(i, stateCurrent);
+    auto rCurrent = stateCurrent.getSubStateHandle<R3>(0);
+    auto currentState = rCurrent.getValue();
+    
+    traj->evaluate(i+1, stateNext);
+    auto rNext = stateNext.getSubStateHandle<R3>(0);
+    auto nextState = rNext.getValue();
+
+    trajDistance += sqrt(pow(nextState[0] - currentState[0], 2) + pow(nextState[1] - currentState[1], 2) + pow(nextState[2] - currentState[2], 2));
+  }
+
+  double simplifiedTrajDistance = 0.0;
+  for(size_t i = 0; i < simplifiedTraj->getDuration(); ++i)
+  {
+    simplifiedTraj->evaluate(i, stateCurrent);
+    auto rCurrent = stateCurrent.getSubStateHandle<R3>(0);
+    auto currentState = rCurrent.getValue();
+
+    simplifiedTraj->evaluate(i+1, stateNext);
+    auto rNext = stateNext.getSubStateHandle<R3>(0);
+    auto nextState = rNext.getValue();
+
+    simplifiedTrajDistance += sqrt(pow(nextState[0] - currentState[0], 2) + pow(nextState[1] - currentState[1], 2) + pow(nextState[2] - currentState[2], 2));
+  }
```

Comment:
Put this code in a helper function, since it is duplicated twice. We can either leave that function in this file in an anonymous `namespace` _or_ put it in Aikido proper. If you choose the latter, please also write tests for that function. 😁  😈 

As a result, we should add a second test that verifies that the return value is `false` if we pass the function a two-waypoint trajectory.